app/predacons/src/generate.py

```
@@ -61,20 +62,64 @@ def __generate_output(model_path, sequence, max_length,trust_remote_code=False,g
+            inputs = tokenizer(sequence, return_tensors="pt", add_special_tokens=False)
+            inputs = {key: tensor.to(model.device) for key, tensor in inputs.items()}
+            streamer = TextIteratorStreamer(tokenizer, skip_prompt=True, skip_special_tokens=True)
+            generation_config = GenerationConfig(
+                temperature=temperature,
+                do_sample=True,
+            )
+            generation_kwargs = dict(inputs, streamer=streamer, max_new_tokens=max_length, generation_config=generation_config)
+            thread = Thread(target=model.generate, kwargs=generation_kwargs)
+            return thread, streamer
+        except Exception as e:
+            raise RuntimeError(f"Failed to setup streaming generation: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)